bundles/io/org.eclipse.smarthome.io.net.security/src/main/java/org/eclipse/smarthome/io/net/security/internal/ServiceConfiguration.java

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.net.security.internal;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.Base64.Encoder;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Configuration of the {@link NetworkServerTlsProvider}.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class ServiceConfiguration {
+    // Multi-context data, needs to be parsed
+    String domains = "";
+    String psks = "";
+    String keystorePasswords = "";
+    String privateKeyPasswords = "";
+
+    // Self-signed certificate configuration data
+    public String organizationalUnit = "Eclipse Smarthome";
+    public String organization = "Eclipse Foundation, Inc.";
+    public String city = "Ottawa";
+    public String state = "Ontario";
+    public String country = Locale.getDefault().getCountry();
+    public BigDecimal validityInDays = BigDecimal.valueOf(365 * 3);
+
+    // Algorithm specific for new RSA or EC public/key pair
+    public String algorithmTLS = "RSA";
+    public String algorithmDTLS = "EC";
+    public String algorithmSigningTLS = "SHA1WithRSA";
```

Comment:
This is the only one that is widespread useable. Because oracle, bound to some stupid US export laws, does not ship its java jdk/jre with all nowadays algorithms, you need the Java Cryptography Extension (JCE) http://www.oracle.com/technetwork/java/javase/downloads/jce8-download-2133166.html

Reply:
So why don't we require that the JCE is installed if one wants to use the new SSL/DTLS service?

One other option would be to use [Bouncy Castle](https://www.bouncycastle.org/java.html) but I am not sure how these US export restrictions apply...

From a security perspective its a real bad idea to use SHA1. Personally I rather do not have a SSL service than one using broken cryptographic algorithms...

Reply:
The algorithms are configurable, so everybody is free to fine tune. I would go with a feature test. The user is able to specify multiple algorithms, highest to lowest priority. If nothing matches, we throw an exception and stop the service (which stops all depending services), so that the user immediately notice that his configuration does not work and we are never in an insecure state.

Reply:
That highly depends on how you define "insecure state"... I would say that a default configuration with SHA1 **is** insecure. We cannot expect from a normal user to choose an encryption mechanism, so we should set a secure default algorithm.